esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -52,7 +52,48 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
     }
     virtual ~CEsdlSDSStore() { }
 
-    virtual void fetchDefinition(const char* definitionId, StringBuffer& esxdl) override
+    virtual IPropertyTree* fetchDefinition(const char* definitionId) override
```

Comment:
Can we common up the code with in those 3 fetchDefinition functions? For example, you can create 2 private functions fetchDefinitionById and fetchDefintionByName that can be called by those 3 functions, or make both "XML" functions call the fetchDefinition function. 
Also in the fetchDefinition function, it seems the "definitionId" can be either ID or name, should we change the parameter name to "defintionIDorName", or simply "definition"?

Reply:
@rpastrana since you're refactoring the code, it'll be a good time to make the coding look better, if the changes are not in a super hurry to release...

Reply:
@mayx I suggest opening a jira for the work you're mentioning. We have to prevent scope creep.